langchain/vectorstores/base.py

```diff
@@ -324,33 +333,17 @@ def validate_search_type(cls, values: Dict) -> Dict:
         """Validate search type."""
         if "search_type" in values:
             search_type = values["search_type"]
-            if search_type not in ("similarity", "mmr"):
+            if search_type not in ("similarity", "similarity_with_score", "mmr"):
                 raise ValueError(f"search_type of {search_type} not allowed.")
         return values
 
     def get_relevant_documents(self, query: str) -> List[Document]:
-        if self.search_type == "similarity":
-            docs = self.vectorstore.similarity_search(query, **self.search_kwargs)
-        elif self.search_type == "mmr":
-            docs = self.vectorstore.max_marginal_relevance_search(
-                query, **self.search_kwargs
-            )
-        else:
-            raise ValueError(f"search_type of {self.search_type} not allowed.")
-        return docs
+        return self.vectorstore.search(query, self.search_type, **self.search_kwargs)
```

Comment:
I felt that it was redundant, so I changed it to use the search method of VectorStore directly.